### ISSUE TITLE
Sensitive information exposed as resource Id - mongodbatlas_x509_authentication_database_user 

### ISSUE THREAD

**Issue by houssemba** [mongodb/terraform-provider-mongodbatlas#884]
### Terraform CLI and Terraform MongoDB Atlas Provider Version

```hcl
Terraform v1.2.5
```

### Terraform Configuration File 

```hcl
resource "mongodbatlas_x509_authentication_database_user" "x509_authentication_database_user" {
  for_each = local.users

  project_id = mongodbatlas_database_user.users[each.key].project_id
  username   = mongodbatlas_database_user.users[each.key].username

  months_until_expiration = 12
}
```

### Steps to Reproduce

- `terraform init`
- `terraform apply`

### Expected Behavior

The created resource should not include the user's certificate in the id. I think that the id should only contains the projectId and the username. By including the certificate and the private key, we take the risk to leak/expose the user's certificate/private key in the logs (in the CI for example when we apply or plan) and in the terraform state (even if it's flagged as sensitive information)

### Actual Behavior

The user's current certificate is used to compute the resource id and found exposed in the logs: 

```
	d.SetId(encodeStateID(map[string]string{
		"project_id":          projectID,
		"username":            username,
		"current_certificate": currentCertificate,
	}))
```

https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/mongodbatlas/resource_mongodbatlas_x509_authentication_database_user.go#L226

**Comment by martinstibbe** [comment]
@houssemba validating issue

**Comment by Zuhairahmed** [comment]
hi @houssemba thanks for raising this issue. as an update we are exploring internally on how to find best path forward. will revert back when we have something to share. feel free to reach out in the interim if anything else comes up here.

**Comment by Zuhairahmed** [comment]
Thanks @houssemba for the feedback. As an update we added this item to our backlog and expect an update over next few quarters. Will keep you posted. Thank you. Internal ticket [INTMDB-468](https://jira.mongodb.org/browse/INTMDB-468).

**Comment by Zuhairahmed** [comment]
@houssemba as an update, this issue has been picked up in current sprint cycle and you can expect fix to be included as part of v1.8.0 release next week. will let you know when latest release has been published to the Terraform Registry.

**Comment by Zuhairahmed** [comment]
@houssemba we just released v1.8.0, feel free to give it try! issue should have been resolved.

**Comment by Zuhairahmed** [comment]
closing this issue, but feel free to re-open if you need anything else
